Add a makefile showing how to run lint and tests, and fix lint

mthaddon opened the pull request:
Add a makefile showing how to run lint and tests, and fix lint

jameinel commented on Makefile:
```diff
@@ -0,0 +1,7 @@
+test: lint
+	@python3 -m unittest
+
+.PHONY: test
```
isn't 'lint' also PHONY?

mthaddon replied:
Thanks, have added that, and made test the default target. I agree having this be authoritative would be ideal. I'm not sure about whether "test" or "check" is more common practice for a standard Makefile target - I've seen both but would be happy to go with either.

jameinel commented on ops/jujuversion.py:
```diff
@@ -38,6 +38,18 @@ def __eq__(self, other):
+    def _less_than_property(self, other, prop):
+        self_prop = getattr(self, prop)
+        other_prop = getattr(other, prop)
+        if self_prop != other_prop:
```
I think this reads a little better as:

```
if self_prop == other_prop:
  return None
if prop == 'tag':
  if not self_prop:
    return False
  if not other_prop:
    return True
return self_prop < other_prop
```

mthaddon replied:
Agreed, have updated in more recent commit.

jameinel commented on Makefile:
```diff
@@ -0,0 +1,7 @@
+	@python3 -m unittest
+
+lint:
+	@flake8
```
Looking at our .travis.yml we'd want to have lint be both:
    - autopep8 -r --aggressive --diff --exit-code .
    - flake8

(IIRC, autopep8 gives you a nicer diff suggestion when your syntax is incorrect, rather than just telling you it is wrong).

And then we can change the .travis.yml script to be:

script:
  make test

I don't know if we'd want to be cautious and add "sudo apt install flake8 make" just to be extra cautious that Make is available on the build machines.

jameinel replied:
```suggestion
	@autopep8 -r --aggressive --diff --exit-code .
	@flake8
```

(testing out github's suggestion feature, I don't seem to be able to suggest changes to a file that isn't already in the PR to suggest the changes to .travis.yml)

mthaddon commented:
Have updated per suggestions

niemeyer commented on ops/jujuversion.py:
```diff
@@ -46,18 +58,8 @@ def __lt__(self, other):
+        for prop in ['major', 'minor', 'tag', 'patch', 'build']:
+            lt = self._less_than_property(other, prop)
+            if lt is not None:
+                return lt
```
The original version had 14 lines.. the new version has 14 lines, except it jumps across two different functions back and forth repeatedly, uses more indirection, and is harder to read. It's also a bit of a lie as it pretends to be general until it's not, by special casing the handling of "tag".

I'd recommend preserving the existing function.

mthaddon commented:
I've reverted to the pre-existing __lt__ code in jujuversion.py, as I agree in retrospect this isn't a great readability improvement. The reason I'd taken this approach was that I was getting a complexity error when running flake8 locally, but that's due to me having ~/.config/flake8, so I've updated this PR to be explicit about the location of the flake8 config to use.
